roles/ansible/tower/manage-workflow-templates/tasks/process-workflow-node.yml

```diff
@@ -0,0 +1,77 @@
+---
+
+- name: "Find id of current job"
+  set_fact:
+    job_template_id: "{{ job_template.id }}"
+  when:
+    - job_template.name == workflow_node.job_name
+  with_items:
+    - "{{ existing_job_templates_output.rest_output }}"
+  loop_control:
+    loop_var: job_template
+
+- block:
```

Comment:
These 3 blocks are pretty much a repeat with just a slight difference - i.e.: the url for the API call. Can we look to combine to avoid repeat? One suggestion is to utilize the `node_type` as mentioned in the comment to the `.py` file above. If this is kept `success_nodes` and `failure_nodes`, you can use that in the composition of the API path and hence combine at least the 2 for success/failure node processing. With a `set_fact` up front, you may be even able to combine all 3 into one.